src/main/java/uk/gov/hmcts/darts/common/service/impl/DataAnonymisationServiceImpl.java

```diff
@@ -97,8 +122,51 @@ public UserAccountEntity getUserAccount() {
         return userIdentity.getUserAccount();
     }
 
+    void tidyUpTransformedMediaEntities(UserAccountEntity userAccount, CourtCaseEntity courtCase) {
+        courtCase
+            .getHearings()
+            .stream()
+            .map(HearingEntity::getMediaRequests)
+            .flatMap(Collection::stream)
+            .distinct()
+            .peek(mediaRequestEntity -> expiredMediaRequest(userAccount, mediaRequestEntity))
+            .map(MediaRequestEntity::getTransformedMediaEntities)
+            .flatMap(Collection::stream)
+            .distinct()
+            .forEach(this::deleteTransformedMediaEntity)
+        ;
+    }
+
+    void deleteTransformedMediaEntity(TransformedMediaEntity transformedMediaEntity) {
+        if (transformedMediaEntity.getTransientObjectDirectoryEntities().stream()
+            .peek(transientObjectDirectoryEntity -> transientObjectDirectoryEntity.setTransformedMedia(null))
+            .filter(transientObjectDirectoryEntity -> !deleteTransientObjectDirectoryEntity(transientObjectDirectoryEntity))
+            .count() == 0) {
```

Comment:
noneMatch might be better than `.count() == 0`?

Reply:
I don't think this works in this scenario as noneMatch and allMatch will stop processing once there criteria has been invalidated which means it will could fail to delete some entries

Reply:
yeah, thats fine.

Reply:
Agreed updated